core_backend/app/llm_call/parse_input.py

```diff
@@ -2,75 +2,196 @@
 These are functions that can be used to parse the input questions.
 """
 
+from functools import wraps
+from typing import Any, Callable, Tuple
+
 from ..configs.llm_prompts import (
     PARAPHRASE_INPUT,
     TRANSLATE_INPUT,
     IdentifiedLanguage,
     SafetyClassification,
 )
-from ..schemas import UserQueryRefined
+from ..schemas import ResultState, UserQueryRefined, UserQueryResponse
 from .utils import _ask_llm
 
 
-def classify_safety(query_text: str) -> SafetyClassification:
+def classify_safety(func: Callable) -> Callable:
+    """
+    Decorator to classify the safety of the question.
+    """
+
+    @wraps(func)
+    async def wrapper(
+        question: UserQueryRefined,
+        response: UserQueryResponse,
+        *args: Any,
+        **kwargs: Any,
+    ) -> UserQueryResponse:
+        """
+        Wrapper function to classify the safety of the question.
+        """
+        question, response = _classify_safety(question, response)
```

Comment:
Note that every function accepts these two input and produces the same two output. This consistency will make it easy to make future extension. Unfortunately (unlike typescript!) there is no easy way to enforce this interface on functions so we'll have to do it by setting norms.